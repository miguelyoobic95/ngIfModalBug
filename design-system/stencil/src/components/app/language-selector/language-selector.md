---
name: Language Selector
category: App Components
---

```yoo-language-selector.html
<yoo-button onclick="generateLanguageModal()">
    <div class="icon">
        <i class="yo-world"></i>
    </div>
</yoo-button>
<yoo-modal-controller></yoo-modal-controller>
```

```yoo-language-selector.js
    function generateLanguageModal() {
        let modalCtrl = document.querySelector('yoo-modal-controller');
        let htmlElement = document.createElement('div');
        htmlElement.innerHTML = `<div >
            <yoo-language-selector></yoo-language-selector>
        </div>`
        modalCtrl.generateModal({
            hasHeader: true,
            hasFooter: false,
            content: htmlElement,
            cssClass: 'yoo-modal-language'
        });
        modalCtrl.show();

        let languageSelector = htmlElement.querySelector('yoo-language-selector');
        languageSelector.languages = [
            {title: 'English', value: 'EN', icon: './assets/flags/united-kingdom.svg'},
        ];
    }
```

## Attributes

|Name|Type|Default|Description|
|---|---|---|---|
|`languages`|Language Object|    |the list to fill the language selection|

## CSS

|Type|Name|Description|
|---|---|---|
|Accent|`accent`|yoo-language-selector accent|
|Dark|`dark`|yoo-language-selector dark|
|Success|`success`|yoo-language-selector success|
|Danger|`danger`|yoo-language-selector danger|
|Info|`info`|yoo-language-selector info|
|Warning|`warning`|yoo-language-selector warning|
|Gradient Accent|`gradient-accent`|yoo-language-selector gradient-accent|
|Gradient Dark|`gradient-dark`|yoo-language-selector gradient-grey|
|Gradient Success|`gradient-success`|yoo-language-selector gradient-success|
|Gradient Danger|`gradient-danger`|yoo-language-selector gradient-danger|
|Gradient Info|`gradient-info`|yoo-language-selector gradient-info|
|Gradient Warning|`gradient-warning`|yoo-language-selector gradient-warning|

## Events
|Attr|Description|
|---|---|---|
|`languageSelected`|event emitted a different language is selected|
