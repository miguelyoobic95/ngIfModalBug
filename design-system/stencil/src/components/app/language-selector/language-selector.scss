:host {
  opacity: 0;
    .heading {
      padding-top: 1.5625rem;
      padding-left: 1.5625rem;
      padding-bottom: 1.0625rem;
      line-height: normal;
      text-transform: uppercase;    
    }
  .load-container {
    width: 100%;
    height: 100%;
  }
  .lists-container {
    padding: 0 1.5625rem;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-between;
    .item {
      width: 2rem;
      height: 3rem;
      align-items: center;
      margin-right: 0.75rem;
      margin-bottom: 0.75rem;
      justify-content: space-between;
      .icon {
        display: flex;
        position: relative;
        border-radius: 1rem;
        height: 2rem;
        width: 2rem;
        justify-content: center;
        align-items: center;
        overflow: hidden;
        font-size: 2rem;
          .icon-class {
            position: absolute;
          }
        }
        .overlay {
          position: absolute;
          width: 2rem;
          height: 2rem;
          top: 0%;
          background-color: cssvar(text-color);
          opacity: 0.6;
          border-radius: 1rem;
        }
        .check {
          position: absolute;
          top: 0.5rem;
          left: 0.5rem;
          background-color: cssvar(light);
          width: 1rem;
          height: 1rem;
          border-radius: 0.5rem;
          display: flex;
          justify-content: center;
          align-items: center;
          font-size: 0.75rem;
        } 
      }
      .item-title {
        font-size: 0.5625rem;
      }
    }
  }
      
