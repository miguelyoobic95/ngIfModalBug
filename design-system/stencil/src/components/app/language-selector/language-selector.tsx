import { Component, EventEmitter, Event, Element, Prop, State } from '@stencil/core';
import { ILanguage } from '@shared/interfaces';
import { setAnimation } from '../../../utils/anim';

@Component({
    tag: 'yoo-language-selector',
    styleUrl: 'language-selector.scss',
    scoped: true
})
export class YooLanguageSelectorComponent {

    @Prop() languages: ILanguage[];
    @Prop() currentLanguage: string;

    @Event() languageSelected: EventEmitter<string>;

    @State() loaded: boolean = false;

     @Element() host: HTMLStencilElement;


    componentDidLoad() {
      setAnimation('fade', this.host, {open: true});
      setTimeout(() => {
        this.loaded = true;
      }, 500);
    }

    onLanguageSelector(language: string) {
        this.languageSelected.emit(language);
        document.querySelector('yoo-modal-controller').closeModal(false);
    }

    renderList(language: ILanguage): JSX.Element {
        return (
          <div class="item" attr-layout="column">
            <div class={'icon ' + (this.currentLanguage === language.value ? 'current' : '')} onClick={() => this.onLanguageSelector(language.value)}>
                <i class={language.icon + ' icon-class'}></i>
                {this.currentLanguage.toLowerCase() === language.value ? [
                <div class="overlay">
                </div>,
                <div class="check">
                <i class="yo-check"></i>
                </div> ]
                : null }
            </div>
            <div class="item-title">
              {language.title}
            </div>
          </div>
        );
    }

    render() {
      return (
        <yoo-modal has-header="false" class="language-selector" >
          <div class="heading">Language</div>
          {this.loaded ?
          <div class="lists-container" attr-layout="row">
            {this.languages.map((language) =>
             this.renderList(language)
            )}
          </div> :
          <div class="load-contaienr" attr-layout="row" attr-layout-align="center center">
            <yoo-loader class="large"></yoo-loader>
          </div>}
        </yoo-modal>
      );
    }
}
