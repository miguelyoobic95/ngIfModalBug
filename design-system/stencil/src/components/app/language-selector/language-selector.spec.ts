import { TestWindow } from '@stencil/core/testing';
jest.mock('../../../utils/anim');
import { YooLanguageSelectorComponent } from './language-selector';

describe('YooLanguageSelectorComponent', () => {
    const data = [{ title: 'English', value: 'EN', icon: 'flag-icon flag-icon-gb' },
    { title: 'English US', value: 'EN', icon: 'flag-icon flag-icon-us' },
    { title: 'Spanish', value: 'ES', icon: 'flag-icon flag-icon-es' },
    { title: 'French', value: 'FR', icon: 'flag-icon flag-icon-fr' },
    { title: 'Dutch', value: 'NL', icon: 'flag-icon flag-icon-nl' },
    { title: 'Italian', value: 'IT', icon: 'flag-icon flag-icon-it' },
    { title: 'German', value: 'DE', icon: 'flag-icon flag-icon-de' }];

    let window;
    beforeEach(() => {
        window = new TestWindow();
    });

    it('should build', () => {
        expect(new YooLanguageSelectorComponent()).toBeTruthy();
    });

    describe('Snapshots', () => {
        it('should render', async () => {
            let element = await window.load({
                components: [YooLanguageSelectorComponent],
                html: '<yoo-language-selector></yoo-language-selector>'
            });
            expect(element).toMatchSnapshot();
        });
        it('should have a current language', async () => {
            let element = await window.load({
                components: [YooLanguageSelectorComponent],
                html: '<yoo-language-selector class="success" current-language="en"></yoo-language-selector>'
                });
            expect(element).toMatchSnapshot();
        });

        it('should have a list of 6 languages', async () => {
            let element = await window.load({
                components: [YooLanguageSelectorComponent],
                html: '<yoo-language-selector class="success" current-language="en"></yoo-language-selector>'
                });
            element.languages = data;
            await window.flush(element);
            expect(element).toMatchSnapshot();
        });

    });
    //TODO fix event
    describe('events', () => {
        xit('should emit on li click', async () => {
            let element = await window.load({
                components: [YooLanguageSelectorComponent],
                html: '<yoo-language-selector></yoo-language-selector>'
            });
            element.languages = data;
            await window.flush();

            let res;
            let emitLi = jest.fn(ev => { res = ev.detail; });
            window.document.addEventListener('languageSelected', emitLi);
            let liItem = element.querySelector('.icon');
            await liItem.click();
            expect(emitLi).toHaveBeenCalled();
            expect(res).toEqual('EN');
        });
    });
});
