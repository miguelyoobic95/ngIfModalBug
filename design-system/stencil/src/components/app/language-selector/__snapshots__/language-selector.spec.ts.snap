// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooLanguageSelectorComponent Snapshots should have a current language 1`] = `
<yoo-language-selector
  class="success hydrated"
  current-language="en"
  data-yoo-language-selector-host=""
>
  
  <yoo-modal
    class="language-selector"
    data-yoo-language-selector=""
    has-header="false"
  >
    <div
      class="heading"
      data-yoo-language-selector=""
    >
      Language
    </div>
    <div
      attr-layout="row"
      attr-layout-align="center center"
      class="load-contaienr"
      data-yoo-language-selector=""
    >
      <yoo-loader
        class="large"
        data-yoo-language-selector=""
      />
    </div>
  </yoo-modal>
</yoo-language-selector>
`;

exports[`YooLanguageSelectorComponent Snapshots should have a list of 6 languages 1`] = `
<yoo-language-selector
  class="success hydrated"
  current-language="en"
  data-yoo-language-selector-host=""
>
  
  <yoo-modal
    class="language-selector"
    data-yoo-language-selector=""
    has-header="false"
  >
    <div
      class="heading"
      data-yoo-language-selector=""
    >
      Language
    </div>
    <div
      attr-layout="row"
      attr-layout-align="center center"
      class="load-contaienr"
      data-yoo-language-selector=""
    >
      <yoo-loader
        class="large"
        data-yoo-language-selector=""
      />
    </div>
  </yoo-modal>
</yoo-language-selector>
`;

exports[`YooLanguageSelectorComponent Snapshots should render 1`] = `
<yoo-language-selector
  class="hydrated"
  data-yoo-language-selector-host=""
>
  
  <yoo-modal
    class="language-selector"
    data-yoo-language-selector=""
    has-header="false"
  >
    <div
      class="heading"
      data-yoo-language-selector=""
    >
      Language
    </div>
    <div
      attr-layout="row"
      attr-layout-align="center center"
      class="load-contaienr"
      data-yoo-language-selector=""
    >
      <yoo-loader
        class="large"
        data-yoo-language-selector=""
      />
    </div>
  </yoo-modal>
</yoo-language-selector>
`;
